judge/views/submission.py

```diff
@@ -245,7 +245,7 @@ def _get_queryset(self):
             if not self.request.user.has_perm('judge.see_private_contest'):
                 # Show submissions for any contest you can edit or visible scoreboard
                 contest_queryset = Contest.objects.filter(Q(organizers=self.request.profile) |
```

Comment:
I think you also need a clause to check for `Participation` visibility contests where your participation has ended, cc @Xyene @quantum5

Reply:
Discussed with quantum in Slack, we should not run such a query.